runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts

```diff
@@ -2,147 +2,150 @@ import { ActionReducerMapBuilder } from '@reduxjs/toolkit';
 
 import { UsersState } from './Users.state';
 import {
-    getAll,
-    getAllNotActivatedByPage,
-    getAllActivatedByPage,
-    getAllActivatedByPageAndTenant,
-    getAllNotActivatedByPageAndTenant,
-    updateNotActivated,
-    updateActivated,
     partialUpdate,
     deleteUser,
-    getActiveNonAdmins,
-    updateNotActivatedByTenant,
-    updateActivatedByTenant,
+    getAllUsersInTenant,
+    getAllByPage,
+    getAllByPageInTenant,
+    update,
+    updateInTenant,
 } from './Users.thunks';
 
+// eslint-disable-next-line max-lines-per-function
 const buildUsersExtraReducers = (builder: ActionReducerMapBuilder<UsersState>) => {
     builder
-        // GET ALL
-        .addCase(getAll.pending, (state) => {
-            state.loading = true;
-        })
-        .addCase(getAll.fulfilled, (state, action) => {
-            state.loading = false;
-            state.all = action.payload;
-        })
-        .addCase(getAll.rejected, (state) => {
-            state.loading = false;
-        })
+        // FOR ADMINS
+        .addCase(getAllByPage.pending, (state, action) => {
 
-        // PARTIAL ACCOUNT UPDATE
-        .addCase(partialUpdate.pending, (state) => {
-            state.userUpdate.loading = true;
-        })
-        .addCase(partialUpdate.fulfilled, (state) => {
-            state.userUpdate.loading = false;
-        })
-        .addCase(partialUpdate.rejected, (state) => {
-            state.userUpdate.loading = false;
-        })
+            if (
+                action.meta.arg?.filter?.equals &&
+                'activated' in action.meta.arg.filter.equals
```

Comment:
to avoid repeating yourself, maybe it is good to extract this type check to function called `hasActivated`?